pmm/pmm2-ui-tests.groovy

```diff
@@ -132,51 +159,63 @@ pipeline {
             }
         }
         stage('Checkout Commit') {
-             when {
+            when {
                 expression { env.GIT_COMMIT_HASH.length()>0 }
             }
             steps {
                 sh 'git checkout ' + env.GIT_COMMIT_HASH
             }
         }
-        stage('Start staging') {
-            steps {
-                runStaging(DOCKER_VERSION, CLIENT_VERSION, '--addclient=ps,1 --addclient=mo,2 --with-replica --addclient=pgsql,1 --addclient=pxc,1 --with-proxysql --pmm2 --setup-alertmanager --add-annotation', CLIENT_INSTANCE, SERVER_IP)
+        stage('Start PMM Server') {
+            when {
+                expression { env.CLIENT_INSTANCE == "no" }
             }
-        }
-        stage('Sanity check') {
             steps {
-                sh 'timeout 100 bash -c \'while [[ "$(curl -s -o /dev/null -w \'\'%{http_code}\'\' \${PMM_URL}/ping)" != "200" ]]; do sleep 5; done\' || false'
+                runStagingServer(DOCKER_VERSION, CLIENT_VERSION, '', CLIENT_INSTANCE, SERVER_IP)
             }
         }
-        stage('Setup Node') {
+        stage('Start PMM Client Instance') {
             steps {
-                sh """
-                    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash
-                    . ~/.nvm/nvm.sh
-                    nvm install 12.14.1
-                    sudo rm -f /usr/bin/node
-                    sudo ln -s ~/.nvm/versions/node/v12.14.1/bin/node /usr/bin/node
-                    pushd pmm-app/
-                    npm install
-                    node -v
-                    npm -v
-                    sudo yum install -y gettext
-                    envsubst < env.list > env.generated.list
-                    popd
-                """
+                runStagingClient(DOCKER_VERSION, CLIENT_VERSION, '--addclient=ps,1 --addclient=mo,2 --with-replica --addclient=pgsql,1 --addclient=pxc,1 --with-proxysql --pmm2 --setup-alertmanager --add-annotation --setup-replication-ps-pmm2', 'yes', env.SERVER_IP)
             }
         }
-        stage('Sleep') {
-            steps {
-                sh """
-                curl --data '{"enable_stt": true, "enable_telemetry": true}' -u admin:admin -X POST ${PMM_UI_URL}/v1/Settings/Change
-                curl -u admin:admin -X POST ${PMM_UI_URL}/v1/management/SecurityChecks/Start
-                """
-                sleep 300
-                sh """
-                curl --data '{"disable_stt": true, "enable_telemetry": true}' -u admin:admin -X POST ${PMM_UI_URL}/v1/Settings/Change
-                """
+        stage('Setup') {
+            parallel {
```

Comment:
Is it really needed to run this block in parallel? For example, just in case if sanity check fails, the pipeline will run till it finishes the whole stage. 

Other parts is good to me

Reply:
yes, but it will still be very rare as compared to the number of jobs which will save 2-3 mins from this in parallel, but if you think it is not needed, I can move sanity Check out of this parallel block

Reply:
Got it, LGTM